### PR TITLE
feat: upgrade faraday dependency

### PR DESCRIPTION
**Pull request by joeldodge79**
- new interface for middleware
-- https://lostisland.github.io/faraday/middleware/custom
- as of version 0.1.0, ruby 2.5 is the oldest version we support

fixes #99

### CODE REVIEW

File: lib/looker-sdk/response/raise_error.rb [hunk 1]
```diff
@@ -30,9 +30,7 @@ module LookerSDK
   module Response
 
     # HTTP status codes returned by the API
-    class RaiseError < Faraday::Response::Middleware
-
-      private
```
Comment (joeldodge79) [inline lib/looker-sdk/response/raise_error.rb:35]: new middleware interface: needs to sublcass `Faraday::Middleware` and publicly expose `on_complete`

File: lib/looker-sdk/version.rb [hunk 2]
```diff
@@ -26,6 +26,6 @@ module LookerSDK
 
   # Current version
   # @return [String]
-  VERSION = "0.0.8"
+  VERSION = "0.1.0"
```
Comment (joeldodge79) [inline lib/looker-sdk/version.rb:29]: feels like dropping old ruby version supports warrants a "pre-major breaking change version bump"

File: Makefile [hunk 3]
```diff
@@ -31,10 +31,8 @@
 # Automatic variables: http://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_chapter/make_10.html#SEC101
 
 # Rbenv-friendly version identifiers for supported Rubys
-20_version = 2.0.0-p648
-21_version = 2.1.10
-23_version = 2.3.1
```
Comment (joeldodge79) [inline Makefile:36]: rbenv won't even install these on my machine anymore

**Review by jkaster — APPROVED** [review]
LGTM

**Pull request merged**